pkg/startup/startup.go

```diff
@@ -720,6 +723,53 @@ func configureIPPools(ctx context.Context, client client.Interface) {
 		return
 	}
 
+	// If CIDRs weren't specified through the environment variables, check if they're present in kubeadm's
+	// config map.
+	if len(ipv4Pool) == 0 || len(ipv6Pool) == 0 {
+		clientset, err := createKubernetesClient()
+		if err == nil {
+			kubeadmConfig, err := clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(KubeadmConfigConfigMap, metav1.GetOptions{})
+			if err != nil {
+				log.WithError(err).Warn("failed to query kubeadm's config map")
+			} else {
+				// Look through the config map for lines starting with 'podSubnet', then assign
+				// the right variable according to the IP family of the matching string.
+				re := regexp.MustCompile(`podSubnet: (.*)`)
```

Comment:
For my own understanding, with dual-stack this configmap will have multiple `podSubnet: ` lines in it?

Reply:
No, from what I could find in related github issues it will be a single line with CIDRs separated by a comma. For example, see https://github.com/kubernetes/kubernetes/pull/82473